src/cogitate.py

```diff
@@ -1,58 +1,312 @@
-"""Command Line Interface for interacting with Github repository info."""
+"""Command Line Interface for the Cogitate tool."""
 
-# from data_collection import collect_commits
 import argparse
-
-# from pprint import pprint
-
-# from driller import find_repositories
-
+import os
+import validators
+from progress.bar import IncrementalBar
 import data_collection
+import data_processor
 import json_handler
 
 
-def main():
+# pylint: disable=R0915
+def main(args):
     """Execute the CLI."""
-    args = retrieve_arguments()
+    if args["testwithprintargs"] == "y":
+        for key, value in args.items():
+            print(key, ":", value)
+        return
+    # check for data in json file to catch user error in use of --clearJson
+    check_Json = json_handler.get_dict_from_json_file("raw_data_storage")
+    if len(check_Json) == 0:
+        print("\n\n     Json empty: Collecting raw data from repo\n\n")
+        args["clearJson"] = True
```

Comment:
If I understood this correctly, shouldn't  this portion be nested inside the condition where the user chooses to retrieve current data? It would let them know that there isn't any data to retrieve because the JSON is empty. You'll have to make needed changes to the argument when you do that.